### PR TITLE
 app-interact/lab: Add oneko as alternative to xeyes #142 

### PR DESCRIPTION
**Pull request by mandrei12**

### CODE REVIEW

File: content/chapters/app-interact/lab/content/arena.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+connect(3, {sa_family=AF_UNIX, sun_path=@"/tmp/.X11-unix/X1"}, 20) = 0
+--- SIGALRM {si_signo=SIGALRM, si_code=SI_KERNEL} ---
+```
+
+When running `oneko`, what we will see different from `xeyes` is the `SIGALRM` signal.
+This means tha `oneko` uses a timer, which was set and then expired.
+
+It is easy to confirm the communication between the X server and `oneko`.
+We see that the cat follows our mouse cursor, behaving similar to `xeyes`.
+After running `oneko` under `strace`, we see communication on the Unix socket that is created at the beginning:
```
Comment (teodutu) [inline content/chapters/app-interact/lab/content/arena.md:20]:
```suggestion
When running `oneko`, what differs from `xeyes` is the `SIGALRM` signal.
This means that `oneko` uses a timer, which is periodically set, and then it expires only to be reset again.

Verifying the communication between the X server and `oneko` is easy.
We see that the cat follows our mouse cursor, behaving similarly to `xeyes`.
After running `oneko` under `strace`, we see the communication uses the UNIX socket created at the beginning:
```

**Review by teodutu — CHANGES_REQUESTED** [review]
I reworded 2 paragraphs. You should also explain why that timer is needed. Its purpose is to delay the cat. You can verify this by running `while true; do kill -14 $(pidof oneko)`. You'll notice the cat is much faster in following your mouse. Add this detail to the exercise as a quiz that requires students to "hack" `oneko` themselves similarly to what I did.

**Comment by gabrielmocanu** [comment]
@mandrei12  bump on this

**Review by teodutu — CHANGES_REQUESTED** [review]
Squash your commits into a single one titled like so: `app-interact/lab: ...`

**Review by gabrielmocanu — APPROVED** [review]

**Review by teodutu — APPROVED** [review]

**Pull request merged**